inspirehep/modules/workflows/models.py

```diff
@@ -74,3 +77,26 @@ class WorkflowsPendingRecord(db.Model):
         nullable=False,
     )
     record_id = db.Column(db.Integer, nullable=False)
+
+
+class WorkflowsRecordSources(db.Model):
+
+    __tablename__ = "workflows_record_sources"
+    __table_args__ = (
+        db.PrimaryKeyConstraint('record_id', 'source'),
+    )
+    source = db.Column(db.Text, default="", nullable=False)
+    record_id = db.Column(
+        UUIDType,
+        db.ForeignKey("records_metadata.id", ondelete='CASCADE'),
```

Comment:
I don't see why this requires an N-M model in the case of non ingested records. Even in that case only one workflow record (identified how?) will match to N sources.

But IMHO to wishing to merge rejected record introduces an unnecessary complexity. If they were rejected the first time, they are surely NON-CORE, therefore, if this time we are accepting them, we can simply take the current version. (this was not discussed but I believe such simplification will make our goals more realistic). We can always have rejected records to be instantiated as stub record, thus be stored as regular one...